src/System.Runtime/tests/System/GCTests.netstandard1.7.cs

```diff
@@ -0,0 +1,184 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Runtime;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class GCTests
+    {
+        private const long maxGarbage = 1000;
+        public static void MakeSomeGarbage()
+        {
+            Version vt;
+
+            for (int i = 0; i < maxGarbage; i++)
+            {
+                // Create objects and release them to fill up memory
+                // with unused objects.
+                vt = new Version();
+            }
+        }
+        [Fact]
+        public static void GetGeneration_WeakReference()
+        {
+            var myobj = new Version();
+            MakeSomeGarbage();
+            WeakReference wkref = new WeakReference(myobj);
+            var currentgen = GC.GetGeneration(wkref);
+            myobj = null;
+
+            Assert.True(currentgen >= 0);
```

Comment:
we should probably be asserting here that the generation returned is the same as `myobj`